### PR TITLE
Implemented OP_SWAP & OP_2SWAP opcodes

### PR DESCRIPTION
**Pull request by varun-doshi**
- [x] issue #https://github.com/keep-starknet-strange/shinigami/issues/54
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

Added implementations for `OP_SWAP` & `OP_2SWAP`
- `OP_SWAP` takes the top 2 elements in the stack and swaps them
- - Eg: [1, 2, ...] --> [2, 1, ...]
- `OP_2SWAP` takes the top 2 pairs and swaps them
- - Eg: [4, 3, 2, 1] --> [2, 1, 4, 3]

### CODE REVIEW

File: src/opcodes/tests/test_opcodes.cairo [hunk 1]
```diff
@@ -362,6 +362,107 @@ fn test_op_depth_empty_stack() {
...
+
+    assert!(res, "Execution of step failed");
+    assert_eq!(dstack.len(), 6, "Stack length is not 1");
+
+    let expected_stack = array!["\x06", "\x05", "\x02", "\x01", "\x04", "\x03"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+}
+
+#[test]
+fn test_op_2swap_underflow() {
```
Comment (b-j-roberts) [inline src/opcodes/tests/test_opcodes.cairo:453]: For this test, I think you will want to use the `#[should_panic]` declaration and do one more ` let _ = engine.step();` line to cause the `OP_2SWAP` overflow : https://book.cairo-lang.org/ch10-01-how-to-write-tests.html#checking-for-panics-with-should_panic

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
Nice work! If you could add the #[should_panic] test this'll be good to merge.

File: src/opcodes/tests/test_opcodes.cairo (hunk 1 above)
Comment (varun-doshi) [reply src/opcodes/tests/test_opcodes.cairo:537]: Understood and implemented the same

**Review by b-j-roberts — APPROVED** [review]
Thanks, nice work! 🚀

**Pull request merged**